(exception_type)(func throw 0)